### PR TITLE
[IMP] global_filter: date filtering and automatic user selection

### PR DESCRIPTION
**Pull request by jash-odoo**
Previously, when users selected the month or quarter filter and saved it, the filter would only return data from the current year. This issue has been fixed in this commit.

Merge the month and quarter data filters into a single month/quarter filter. This will allow users to select a year which behave as year filter. If a user only selects a month or quarter, the filter will work accordingly.

The year filter will be removed, as the month/quarter filter can now be used to filter data by year.

In addition, a checkbox will be added to the relation filter that will allow users to automatically select the current user. This checkbox will only be displayed if the model is res.users.

Task ID: 3370627

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr
rresponding to that user.

Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/spreadsheet/static/src/global_filters/helpers.js [hunk 1]
```diff
@@ -144,8 +144,10 @@ export function getRelativeDateDomain(now, offset, rangeType, fieldName, fieldTy
  * @returns {Array<Object>}
  */
 export function dateOptions(type) {
-    if (type === "month") {
-        return monthsOptions;
+    if (type === "monthQuarter") {
+        return [...monthsOptions, ...getPeriodOptions(DateTime.local()).filter(({ id }) =>
+            FILTER_DATE_OPTION["quarter"].includes(id)
+        )];
     } else {
```
Comment (somu-odoo) [inline addons/spreadsheet/static/src/global_filters/helpers.js:151]: Not needed to be put in a if / else case anymore when there is only ONE expected `type`.  Crosscheck it's functionality once and update the `param` accordingly.

**Review by somu-odoo — CHANGES_REQUESTED** [review]
Looks good to me, just couple of changes.

**Review by LucasLefevre — COMMENTED** [review]
Hello,
I already pushed a commit addressing the comments and other things. I'll continue working on this. OXP approaches, we have to move fast, it will be easier from here.
Thanks

**Comment by chapital** [comment]
There's a Way to ADD a TODAY filter, to avoid select a range of date

E.g. 01/01/2023 to 01/01/2023

Also if you have the today filter this gonna show the info of today, and not the range of date saved.

**Review by hokolomopo — COMMENTED** [review]
Olà :wave:

File: addons/spreadsheet/static/src/global_filters/components/filter_date_value/filter_date_value.js [hunk 2]
```diff
@@ -1,15 +1,17 @@
 /** @odoo-module */
 
 import { Component, onWillUpdateProps } from "@odoo/owl";
-import { dateOptions } from "@spreadsheet/global_filters/helpers";
 import { DateTimeInput } from "@web/core/datetime/datetime_input";
+import { FILTER_DATE_OPTION, monthsOptions } from "@spreadsheet/assets_backend/constants";
```
[the remark this answers is not in the document]
Comment (hokolomopo) [reply addons/spreadsheet/static/src/global_filters/components/filter_date_value/filter_date_value.js:5]: We could sort it, we only need it sorted once for the filter_date_value component right ?

File: addons/spreadsheet/static/src/global_filters/helpers.js [hunk 3]
```diff
@@ -22,18 +18,25 @@ export function checkFiltersTypeValueCombination(type, value) {
                     return CommandResult.InvalidValueTypeCombination;
                 }
                 break;
-            case "date":
-                if (typeof value === "string") {
+            case "date": {
+                if (value === "") {
+                    return CommandResult.Success;
```
Comment (LucasLefevre) [inline addons/spreadsheet/static/src/global_filters/helpers.js:23]: this is a bit stupid, but I keep the empty string as a valid value for backward compatibility.

**Pull request closed without merging**